types/jobs.go

```diff
@@ -74,19 +83,43 @@ var (
 	// TwitterAPICaps are basic Twitter capabilities available with API keys
 	TwitterAPICaps = []Capability{CapSearchByQuery, CapGetById, CapGetProfileById}
 
-	// AlwaysAvailableCapabilities defines the job capabilities that are always available regardless of configuration
-	AlwaysAvailableCapabilities = WorkerCapabilities{
-		WebJob:       AlwaysAvailableWebCaps,
-		TelemetryJob: AlwaysAvailableTelemetryCaps,
-		TiktokJob:    AlwaysAvailableTiktokCaps,
-	}
+	// TwitterApifyCaps are Twitter capabilities available with Apify
+	TwitterApifyCaps = []Capability{CapGetFollowers, CapGetFollowing}
 )
 
-// String returns the string representation of the JobType
-func (j JobType) String() string {
-	return string(j)
+// JobCapabilityMap defines which capabilities are valid for each job type
+var JobCapabilityMap = map[JobType][]Capability{
+	// Twitter job types and their valid capabilities
+	TwitterJob: combineCapabilities(
+		TwitterCredentialCaps,
+		TwitterAPICaps,
+		TwitterApifyCaps,
+		[]Capability{CapSearchByFullArchive}, // Elevated API capability
+	),
+	TwitterCredentialJob: TwitterCredentialCaps,
+	TwitterApiJob: combineCapabilities(
+		TwitterAPICaps,
+		[]Capability{CapSearchByFullArchive}, // Elevated API capability
+	),
+	TwitterApifyJob: TwitterApifyCaps,
+
+	// Web job capabilities
+	WebJob: AlwaysAvailableWebCaps,
+
+	// TikTok job capabilities
+	TiktokJob: AlwaysAvailableTiktokCaps,
+
+	// Telemetry job capabilities
+	TelemetryJob: AlwaysAvailableTelemetryCaps,
 }
 
-// WorkerCapabilities represents all capabilities available on a worker
-// Maps JobType to the list of capabilities available for that job type
-type WorkerCapabilities map[JobType][]Capability
+// if no capability is specified, use the default capability for the job type
+var JobDefaultCapabilityMap = map[JobType]Capability{
+	TwitterJob:           CapSearchByQuery,
+	TwitterCredentialJob: CapSearchByQuery,
+	TwitterApiJob:        CapSearchByQuery,
+	TwitterApifyJob:      CapGetFollowers,
+	WebJob:               CapScraper,
+	TiktokJob:            CapTranscription,
+	TelemetryJob:         CapTelemetry,
```

Comment:
This is very nice. At some point we'll need to add to a README somewhere (probably in the tee-indexer repo since it's private) all we need to do to add a new scraper (all the changes needed in tee-worker, tee-indexer and tee-types).